addons/web_editor/models/ir_http.py

```diff
@@ -8,11 +8,14 @@
 class IrHttp(models.AbstractModel):
     _inherit = 'ir.http'
 
-    def _dispatch(self):
-        if 'editable' in request.httprequest.args and 'editable' not in request.context:
-            request.context['editable'] = True
-        if 'edit_translations' in request.httprequest.args and 'edit_translations' not in request.context:
-            request.context['edit_translations'] = True
-        if request.context.get('lang') != "en_US" and 'translatable' not in request.context:
-            request.context['translatable'] = True
-        return super(IrHttp, self)._dispatch()
+    @classmethod
+    def _dispatch(cls):
+        context = dict(request.context)
+        if 'editable' in request.httprequest.args and 'editable' not in context:
+            context['editable'] = True
+        if 'edit_translations' in request.httprequest.args and 'edit_translations' not in context:
+            context['edit_translations'] = True
+        if context.get('lang') != "en_US" and 'translatable' not in context:
+            context['translatable'] = True
+        request.context = context
```

Comment:
`request.context` is a `property` with a setter. The latter stores the context as a `frozendict`.